ngraph/core/reference/include/ngraph/runtime/reference/roll.hpp

```diff
@@ -0,0 +1,61 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/coordinate_transform.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            void roll(const T* arg,
+                      const int64_t* shift,
+                      const int64_t* axes,
```

Comment:
Type of `shift` and `axes` inputs should be a template parameter.
Or may be we can convert these inputs to int64 in the "evaluate" method and have just one implementation? @ilyachur , what do you think?

Reply:
At the current moment I see that here we have a template reference implementation.
I think we can use `int64_t` for axes.
Moreover I think we can avoid any template code in this implementation, because I see that we don't change values (we change the order of elements). It means that we can add element_size as an argument of this method and use memcpy in order to copy data from source to destination.